app/Http/Controllers/Admin/RoleController.php

```diff
@@ -0,0 +1,336 @@
+
+                $string = $string . '&nbsp;&nbsp;' . '<a href="javascript:;" data-id="'. $role->id .'" class="m-portlet__nav-link btn m-btn m-btn--hover-brand m-btn--icon m-btn--icon-only m-btn--pill edit_role" data-tooltip="tooltip" title="'. __('Edit') .'"><i class="la la-edit" aria-hidden="true"></i></a>' . '&nbsp;&nbsp;';
+
+                $string = $string . '&nbsp;&nbsp;' . '<a href="javascript:;" data-id="'. $role->id .'" class="m-portlet__nav-link btn m-btn m-btn--hover-brand m-btn--icon m-btn--icon-only m-btn--pill delete_role" data-tooltip="tooltip" title="'. __('Delete') .'"><i class="la la-trash"></i></a>';
+
+                return $string;
+            })
+
+        ->make(true);
+    }
+
+    /**
+     * [getPermissionRole : Lấy những quyền hạn thuộc vai trò đó]
+     * 
+     * @param  [type] $role_id [description]
+     * @return [type]          [description]
+     */
+    public function getPermissionRole($role_id)
+    {
+        $role = $this->roleRepository->findOrFail($role_id);
+
+        $permissions = $this->permissionRepository->orderBy('id', 'desc')->get();
+
+        return Datatables::of($permissions)
+            ->addIndexColumn()
+
+            ->editColumn('created_at', function($permission){
+                $time = $permission->created_at;
+
+                $time_numb = strtotime($time);
+
+                return date('H:i | d-m-Y', $time_numb);
+            })
+
+            ->addColumn('action', function ($permission) use ($role){
+                $flag = DB::table('permission_role')
+                        ->where('role_id', $role->id)
+                        ->where('permission_id', $permission->id)
+                        ->where('deleted_at', null)
+                        ->first();
+
+                if ($flag != null) {
+                    $string = '<input type="hidden" id="checked-'. $permission->id .'" value="1">';
+
+                    $string = $string . '<i id="action-'. $permission->id. '" class="fa fa-check-circle" data-tooltip="tooltip"  title="Xóa" onclick="updatePermission('. $role->id . ', '. $permission->id .')" aria-hidden="true" style="cursor: pointer; color: #3598dc; font-size: 20px;"></i>';
+                } else {
+                    $string = '<input type="hidden" id="checked-' .$permission->id . '" value="0">';
+
+                    $string = $string . '<i id="action-'. $permission->id .'" class="fa fa-circle-notch" data-tooltip="tooltip" title="Thêm" onclick="updatePermission('. $role->id .', '. $permission->id .')" aria-hidden="true" style="cursor: pointer; color: #3598dc; font-size: 20px;"></i>';
+                }
+
+                return $string;
+            })
+
+        ->make(true);
+    }
+
+    /**
+     * [updatePermissionRole : Cập nhật quyền hạn cho vai trò]
+     * 
+     * @param  Request $request [description]
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/RoleController.php</br>Whitespace found at end of line